### PR TITLE
[#223] fromZecString reject bad input.

### PR DESCRIPTION
**Pull request by HonzaR**
The solution to this issue consists of these steps:
- Implementing regex solution for validation of input ZecAmount after confirm button is pressed.
- Added new unit tests for this new feature.
- Move necessary components from sdk-ext-lib to sdk-ext-ui module.

Implementing the issue, there were filed these new issues:
- https://github.com/zcash/secant-android-wallet/issues/343
- https://github.com/zcash/secant-android-wallet/issues/321
- https://github.com/zcash/secant-android-wallet/issues/322

This pull request closes https://github.com/zcash/secant-android-wallet/issues/223.

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [ ] Manual tests: Did you update the [manual tests](../blob/main/docs/testing/manual_testing) as appropriate? _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._ Code Coverage can be run with: `./gradlew connectedCheck -PIS_COVERAGE_ENABLED=true`
- [ ] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes?
- [ ] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._

# Reviewer

- [x] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) checklist?
- [x] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after changes and when possible, leave it in a better place._
- [x] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Review by ccjernigan — COMMENTED** [review]
Thanks for thorough implementation and filing the additional issues that you identified 👍 

Where are the additional tests?  I see them mentioned in the commit message, but not in the commit itself.

And can we remove the `@Ignore` for some of the existing tests now?

**Comment by HonzaR** [comment]
> Thanks for thorough implementation and filing the additional issues that you identified 👍
> 
> Where are the additional tests? I see them mentioned in the commit message, but not in the commit itself.
> 
> And can we remove the `@Ignore` for some of the existing tests now?

Sorry for that, I somehow mistakenly missed adding them to the commit, although they've been implemented already. Now added also with refactoring to fit the CI testing task.

File: sdk-ext-ui/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 <resources xmlns:tools="http://schemas.android.com/tools" xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
     <string name="zec_amount_regex_continuous_filter" formatted="true" tools:ignore="TypographyDashes">^([0-9]*([0-9]+([<xliff:g id="group" example=",">%1$s</xliff:g>]$|[<xliff:g id="group" example=",">%1$s</xliff:g>][0-9]+))*([<xliff:g id="dec" example=".">%2$s</xliff:g>]$|[<xliff:g id="dec" example=".">%2$s</xliff:g>][0-9]+)?)?$</string>
+    <string name="zec_amount_regex_confirm_filter" formatted="true" tools:ignore="TypographyDashes">^([0-9]{1,3}(?:[<xliff:g id="group" example=",">%1$s</xliff:g>]?[0-9]{3})*)*(?:[0-9]*[<xliff:g id="group" example=".">%2$s</xliff:g>][0-9]*)?$</string>
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply sdk-ext-ui/src/main/res/values/strings.xml:3]: Thanks for pointing this out. I pretty like the Gradle way of prefixing the resources.

File: app/build.gradle.kts [hunk 2]
```diff
@@ -148,6 +148,7 @@ dependencies {
     androidTestImplementation(libs.bundles.androidx.test)
     androidTestImplementation(projects.sdkExtLib)
     androidTestImplementation(projects.spackleLib)
+    androidTestImplementation(projects.sdkExtUi)
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply app/build.gradle.kts:151]: Ok, I haven't noticed that in code. I'll change it.

File: sdk-ext-ui/build.gradle.kts [hunk 3]
```diff
@@ -16,6 +16,8 @@ android {
 dependencies {
     implementation(projects.sdkExtLib)
 
+    implementation(libs.kotlinx.coroutines.android)
```
[the remark this answers is not in the document]
Comment (HonzaR) [reply sdk-ext-ui/build.gradle.kts:19]: Yes, agree.

**Review by ccjernigan — APPROVED** [review]

**Pull request merged**